### PR TITLE
chore: update scalafmt

### PR DESCRIPTION
**Pull request by AlixBa**
When trying to update to sbt-scalafmt to 2.4.4 on projects using Caliban, we get an error on codegen/formatting because one method is deprecated

```
Generating code for schema.graphql
org.scalafmt.dynamic.ScalafmtDynamicError$ScalafmtInterfaceMethodDeprecated: Method Scalafmt.withRespectVersion is deprecated
org.scalafmt.dynamic.ScalafmtDynamic.withRespectVersion(ScalafmtDynamic.scala:62)
org.scalafmt.dynamic.ScalafmtDynamic.withRespectVersion(ScalafmtDynamic.scala:20)
caliban.tools.Formatter$.$anonfun$format$2(Formatter.scala:27)
zio.internal.FiberContext.evaluateNow(FiberContext.scala:490)
zio.internal.FiberContext.$anonfun$evaluateLater$1(FiberContext.scala:776)
java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
java.base/java.lang.Thread.run(Thread.java:832) 
```

### CODE REVIEW

File: tools/src/main/scala/caliban/tools/Formatter.scala [hunk 1]
```diff
@@ -24,8 +24,6 @@ object Formatter {
       val scalafmt =
         Scalafmt
           .create(this.getClass.getClassLoader)
-          .withRespectVersion(false)
-          .withDefaultVersion("2.7.5")                           // For retro-compatibility
```
Comment (ghostdogpr) [inline tools/src/main/scala/caliban/tools/Formatter.scala:28]: @guizmaii was this important?

Comment (guizmaii) [reply tools/src/main/scala/caliban/tools/Formatter.scala:28]: It was important for backward compatibility only

Comment (AlixBa) [reply tools/src/main/scala/caliban/tools/Formatter.scala:28]: FTR it prevents any updates of scalafmt

```
[error] /caliban/tools/src/main/scala/caliban/tools/Formatter.scala:28:12: method withDefaultVersion in trait Scalafmt is deprecated
[error]           .withDefaultVersion("2.7.5")                           // For retro-compatibility
[error]            ^
[error] /caliban/tools/src/main/scala/caliban/tools/Formatter.scala:27:12: method withRespectVersion in trait Scalafmt is deprecated
[error]           .withRespectVersion(false)
[error]            ^
[error] two errors found
[error] (tools / Compile / compileIncremental) Compilation failed

```

Comment (ghostdogpr) [reply tools/src/main/scala/caliban/tools/Formatter.scala:28]: But it looks like it's failing if the user doesn't specify any scalafmt file. We need to support that case (maybe use a fake one where we just add version 2.7.5).

**Comment by AlixBa** [comment]
@ghostdogpr all good

File: tools/src/main/scala/caliban/tools/Formatter.scala (hunk 1 above)
Comment (guizmaii) [reply tools/src/main/scala/caliban/tools/Formatter.scala:28]: > FTR it prevents any updates of scalafmt

I think it's more the dependency of Caliban on `scalafmt-dynamic` that prevents that.

**Review by ghostdogpr — APPROVED** [review]
Thanks!

**Pull request merged**